### ISSUE TITLE
Downstream build fails when using annotation processor and compile options `-Xlint:all -Werror`

### ISSUE THREAD

**Issue by rognan** [jstachio/jstachio#342]
Using the annotation processor in a build along with compiler options `-Xlint:all -Werror` fails the build with:

```sh
warning: No processor claimed any of these annotations:
/io.jstach.jstache.JStache,
... (other annotations which built just fine before jstachio)
/org.junit.jupiter.params.ParameterizedTest,
/org.junit.jupiter.api.io.TempDir

Note: Wrote services file: io.jstach.jstachio.spi.TemplateProvider
Note: Wrote services file: io.jstach.jstachio.spi.JStachioExtension

error: warnings found and -Werror specified
1 error
1 warning
```

Similar issue to the one solved [in project lombok here](https://github.com/projectlombok/lombok/commit/cec8705c3b9fa6488d53bec5bb4eab27a6decf85).

**Comment by agentgt** [comment]
I had originally claimed the annotations but errorprone says that is bad. I may need a flag. I’ll get this fixed for very soon.

**Comment by agentgt** [comment]
https://errorprone.info/bugpattern/DoNotClaimAnnotations

I will check with the errorprone team as well.

I swore I have `-Xlint:all -Werror` in my other projects but maybe I'm mistaken.

**Comment by rognan** [comment]
> I swore I have `-Xlint:all -Werror` in my other projects but maybe I'm mistaken.

Here's a [proof of concept](https://github.com/jstachio/jstachio/files/14796320/poc.zip) build as well as the workaround (Adding `-Xlint:-processing`)

**Comment by rognan** [comment]
> https://errorprone.info/bugpattern/DoNotClaimAnnotations
💯 

I didn't know this was problematic .. 🤷 I'm looking forward to errorprone's reply 👌

**Comment by agentgt** [comment]
Apparently my other project had a flag to disable the warning but for a different reason. 

Would adding an annotation processor flag be amenable for now?

Annotation processors look like `-A` and are covered in JStachio doc.

The issue is I can't add this till like at least a minor release and not a bug release.

The flag I can add as a bug release (well semver is sort of unclear on this because its not actual runtime but lets ignore that) which will not impact people relying on the annotation for whatever reasons.

Also how urgent is this for you?

**Comment by rognan** [comment]
> Would adding an annotation processor flag be amenable for now?
> Also how urgent is this for you?

Using `-Xlint:-processing` works fine for me for now - and if it turns out that errorprone is correct when they say that claiming annotations **prevents** other annotation processors from seeing the annotation, then it's probably something that should always be in effect.

**Comment by agentgt** [comment]
> Using -Xlint:-processing works fine for me for now 

Yes that is the flag I have set in the other project. I will eventually add a flag as I can see an advantage to still have the warning applied to other processors.

Thanks for the bug report!
